.github/workflows/verify-docker-image.yml

```diff
@@ -21,21 +25,34 @@ jobs:
           docker compose version
           npm exec jest -- ./tests/make --runInBand
 
+  build:
```

Comment:
This is probably the most contraversial change. We are building the image and actually pushing it to dockerhub. This means the image is available for anyone to run locally (could be seen as a good thing) especially for testing you can just specify the tag and run the exact latest version of whatever is made in the pr locally. Kind of cool.

I have investigated whether it is possible to remove iamges from dockerhub programatically. It is using their api or a community action so we could utilize that if we don't want images to sit there forever.

Happy to hear feedback on this. It is totally possible to implement these changes without pushing the image. It just means we have to build locally (in the runner I mean) a lot more and we cannot be 100% certain that each job is running with the exact same build. This is the only way to guarantee that.

Reply:
Can we put this in a separate PR to unblock builds now while we think about it ? I'm not sure we want to do that so I'd prefer if we focused on unblocking builds and take the time to think about the implications of pushing an image for every PR.

Reply:
Yep, can do. I'll rollback that part of the PR.

Reply:
Done. I used a force push to keep everything on a single commit.

Reply:
I've kept the version reference as it is using a more "unique" version tag anyway that should drastically reduce likelihood of collisions unless you push an image with the exact PR number, ref and commit hash.